src/send_future.rs

```diff
@@ -163,18 +166,20 @@ impl SendFuture<ActorErasedSending, Broadcast> {
 }
 
 /// The core state machine around sending a message to an actor's mailbox.
-enum Sending<A, Rc: RefCounter> {
+enum Sending<A, M, Rc: RefCounter> {
     New {
-        msg: SentMessage<A>,
+        msg: M,
         sender: inbox::Sender<A, Rc>,
     },
-    WaitingToSend(Arc<spin::Mutex<WaitingSender<A>>>),
+    WaitingToSend(WaitingSender<M>),
     Done,
 }
 
-impl<A, Rc> Future for Sending<A, Rc>
+impl<A, M, Rc> Future for Sending<A, M, Rc>
 where
     Rc: RefCounter,
+    M: Unpin,
```

Comment:
Is this bound still required?

Reply:
Yes but in practice, it doesn't matter because `M` will always be an `Arc` or a `Box` here. There is no `Unpin` bound on the `send` and `broadcast` functions on `Address` so users are not restricted on what they can send!

We could probably add a test for that to our new `public_api` test :)

Reply:
We can remove this bound if we remove the `TrySend` trait and duplicate the `Future` impl here. Would you prefer that?

Reply:
Checkout [`e9eb9d3` (#166)](https://github.com/Restioson/xtra/pull/166/commits/e9eb9d3a32b06195ef0061997b77dd69fcd012ce) and let me know what you prefer :)

Reply:
The duplication is actually not as bad as I expected. I think I prefer the version without `TrySend`.